### ISSUE TITLE
Dubbo is not compatible with Nacos2 client.

### ISSUE THREAD

**Issue by KomachiSion** [apache/dubbo#7291]
Recently, Nacos 2.0 Beta released, and some of users found that dubbo 2.7.X can't be compatible with Nacos2 client.

We found that dubbo 2.7.X get `agent` field by reflection.

I think this logic in dubbo 2.7.X is a beta feature for metadata center, which is not necessary for dubbo 2.7.X.

So I think It should not block the start step.

```
    NacosDynamicConfiguration(URL url) {
        this.nacosProperties = buildNacosProperties(url);
        this.configService = buildConfigService(url);
        this.httpAgent = getHttpAgent(configService);
        watchListenerMap = new ConcurrentHashMap<>();
    }

    private HttpAgent getHttpAgent(ConfigService configService) {
        HttpAgent agent = null;
        try {
            Field field = configService.getClass().getDeclaredField("agent");
            field.setAccessible(true);
            agent = (HttpAgent) field.get(configService);
        } catch (Exception e) {
            throw new IllegalStateException(e);
        }
        return agent;
    }
```

### Environment

* Dubbo version: All 2.7.X
* Operating System version: All OS
* Java version: 1.8

### Steps to reproduce this issue

1. create any dubbo demo project
2. change the nacos dependency to nacos-client 2.0.0-ALPHA/BETA
3. run project and see error

Pls. provide [GitHub address] to reproduce this issue.

### Expected Result

run project successfully

### Actual Result

What actually happens?

If there is an exception, please attach the exception trace:

```
Exception in thread "main" java.lang.IllegalStateException: java.lang.NoSuchFieldException: agent
	at org.apache.dubbo.configcenter.support.nacos.NacosDynamicConfiguration.getHttpAgent(NacosDynamicConfiguration.java:133)
	at org.apache.dubbo.configcenter.support.nacos.NacosDynamicConfiguration.<init>(NacosDynamicConfiguration.java:109)
	at org.apache.dubbo.configcenter.support.nacos.NacosDynamicConfigurationFactory.createDynamicConfiguration(NacosDynamicConfigurationFactory.java:39)
	at org.apache.dubbo.common.config.configcenter.AbstractDynamicConfigurationFactory.lambda$getDynamicConfiguration$0(AbstractDynamicConfigurationFactory.java:39)
	at java.util.concurrent.ConcurrentHashMap.computeIfAbsent(ConcurrentHashMap.java:1660)
	at org.apache.dubbo.common.config.configcenter.AbstractDynamicConfigurationFactory.getDynamicConfiguration(AbstractDynamicConfigurationFactory.java:39)
	at org.apache.dubbo.common.config.configcenter.DynamicConfiguration.getDynamicConfiguration(DynamicConfiguration.java:224)
	at org.apache.dubbo.config.bootstrap.DubboBootstrap.prepareEnvironment(DubboBootstrap.java:882)
	at org.apache.dubbo.config.bootstrap.DubboBootstrap.startConfigCenter(DubboBootstrap.java:605)
	at org.apache.dubbo.config.bootstrap.DubboBootstrap.useRegistryAsConfigCenterIfNecessary(DubboBootstrap.java:674)
	at org.apache.dubbo.config.bootstrap.DubboBootstrap.initialize(DubboBootstrap.java:511)
	at org.apache.dubbo.config.bootstrap.DubboBootstrap.start(DubboBootstrap.java:744)
	at org.apache.dubbo.config.spring.context.DubboBootstrapApplicationListener.onContextRefreshedEvent(DubboBootstrapApplicationListener.java:59)
	at org.apache.dubbo.config.spring.context.DubboBootstrapApplicationListener.onApplicationContextEvent(DubboBootstrapApplicationListener.java:52)
	at org.apache.dubbo.config.spring.context.OneTimeExecutionApplicationContextEventListener.onApplicationEvent(OneTimeExecutionApplicationContextEventListener.java:40)
	at org.springframework.context.event.SimpleApplicationEventMulticaster.doInvokeListener(SimpleApplicationEventMulticaster.java:172)
	at org.springframework.context.event.SimpleApplicationEventMulticaster.invokeListener(SimpleApplicationEventMulticaster.java:165)
	at org.springframework.context.event.SimpleApplicationEventMulticaster.multicastEvent(SimpleApplicationEventMulticaster.java:139)
	at org.springframework.context.support.AbstractApplicationContext.publishEvent(AbstractApplicationContext.java:393)
	at org.springframework.context.support.AbstractApplicationContext.publishEvent(AbstractApplicationContext.java:347)
	at org.springframework.context.support.AbstractApplicationContext.finishRefresh(AbstractApplicationContext.java:883)
	at org.springframework.context.support.AbstractApplicationContext.refresh(AbstractApplicationContext.java:546)
	at org.springframework.context.annotation.AnnotationConfigApplicationContext.<init>(AnnotationConfigApplicationContext.java:84)
	at org.apache.dubbo.samples.ProviderBootstrap.main(ProviderBootstrap.java:32)
Caused by: java.lang.NoSuchFieldException: agent
	at java.lang.Class.getDeclaredField(Class.java:2070)
	at org.apache.dubbo.configcenter.support.nacos.NacosDynamicConfiguration.getHttpAgent(NacosDynamicConfiguration.java:129)
	... 23 more
```

**Comment by AlbumenJ** [comment]
Fixed in 3.0

**Comment by Melod-YI** [comment]
这个问题在2.7.x的版本合入了吗
